components/tidb_query/src/expr/builtin_string.rs

```diff
@@ -914,6 +914,37 @@ impl ScalarFunc {
             .map(|i| 1 + i as i64)
             .or(Some(0)))
     }
+
+    #[inline]
+    // See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_quote
+    pub fn quote<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt_or!(
+            self.children[0].eval_string(ctx, row),
+            Some(Cow::Borrowed(b"NULL"))
+        );
+        let mut result = Vec::<u8>::with_capacity(s.len());
```

Comment:
`s.len()` is not enough. You need `s.len() + 2`or more.

Reply:
I think could calculate the capacity before to avoid the reallocation.